CRM/ManualDirectDebit/BAO/RecurrMandateRef.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class CRM_ManualDirectDebit_BAO_RecurrMandateRef extends CRM_ManualDirectDebit_DAO_RecurrMandateRef {
+
+  /**
+   * @param $params
+   * @param $defaults
+   *
+   * @return \CRM_ManualDirectDebit_BAO_RecurrMandateRef|null
+   */
+  public static function retrieve(&$params, &$defaults) {
+    $object = new CRM_ManualDirectDebit_BAO_RecurrMandateRef();
+    $object->copyValues($params);
+
+    if ($object->find(TRUE)) {
+      CRM_Core_DAO::storeValues($object, $defaults);
+      $object->free();
+      return $object;
+    }
+
+    return NULL;
+  }
+
+  /**
+   * @param $params
+   *
+   * @return \CRM_Core_DAO
+   */
+  public static function add(&$params) {
+    $entity = new CRM_ManualDirectDebit_BAO_RecurrMandateRef();
+    $entity->copyValues($params);
+
+    return $entity->save();
+  }
+
+  /**
+   * @param $params
+   *
+   * @return \CRM_Core_DAO
+   */
+  public static function &create(&$params) {
+    $transaction = new CRM_ManualDirectDebit_BAO_RecurrMandateRef();
```

Comment:
this should be 

```php
$transaction = new CRM_Core_Transaction();
```

instead of 

```php
$transaction = new CRM_ManualDirectDebit_BAO_RecurrMandateRef();
```

though I don't see a reason here to use a transaction, we are only writing to one table here.